### PR TITLE
updated dag version, sha and moab version

### PR DESCRIPTION
**Pull request by shimwell**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Work in progress, do not merge

This PR updates the recipe to use the recently release DAGMC version 3.2.1 and also pins the version of MOAB to 5.3.1

This recipe has been tested locally using the ```python build-locally.py linux_64_``` command as described in the cornda-forge [docs](https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes)

I am currently getting hdf5 conflicts so that is why this PR is not ready to merge

``bash

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.9/site-packages/conda_build/cli/main_build.py", line 488, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.9/site-packages/conda_build/cli/main_build.py", line 477, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 3088, in build_tree
    packages_from_this = build(metadata, stats,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/build.py", line 2051, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 788, in expand_outputs
    for (output_dict, m) in deepcopy(_m).get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 2120, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(ref_metadata, conda_packages, pass_no=0,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/metadata.py", line 781, in finalize_outputs_pass
    fm = finalize_metadata(om, parent_metadata=parent_metadata,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 546, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(m,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 408, in add_upstream_pins
    host_deps, host_unsat, extra_run_specs_from_host = _read_upstream_pin_files(m, 'host',
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 373, in _read_upstream_pin_files
    deps, actions, unsat = get_env_dependencies(m, env, m.config.variant,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/render.py", line 138, in get_env_dependencies
    actions = environ.get_install_actions(tmpdir, tuple(dependencies), env,
  File "/opt/conda/lib/python3.9/site-packages/conda_build/environ.py", line 801, in get_install_actions
    raise DependencyNeedsBuildingError(exc, subdir=subdir)
conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {"libstdcxx-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=7.2.0']", "libgcc-ng[version='>=9.4.0']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=9.4.0|>=7.2.0']", "__glibc[version='>=2.17,<3.0.a0']", "libstdcxx-ng[version='>=9.4.0']", "__glibc[version='>=2.17']", "libgcc-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=7.2.0']", "libstdcxx-ng[version='>=4.9|>=7.3.0|>=7.5.0|>=9.3.0|>=9.4.0|>=7.2.0']", "hdf5[version='>=1.10.5,<1.10.6.0a0|>=1.10.5,<1.10.6.0a0|>=1.10.5,<1.10.6.0a0|>=1.10.6,<1.10.7.0a0|>=1.10.6,<1.10.7.0a0|>=1.10.6,<1.10.7.0a0',build='mpi_openmpi_*|mpi_openmpi_*|mpi_mpich_*']", 'feature:/linux-64::__glibc==2.12=0', 'feature:|@/linux-64::__glibc==2.12=0', "hdf5[version='*|*|*|>=1.12.1,<1.12.2.0a0|>=1.12.1,<1.12.2.0a0|>=1.12.1,<1.12.2.0a0',build='mpi_openmpi_*|mpi_openmpi_*|nompi_*|mpi_mpich_*|mpi_mpich_*']"}
Traceback (most recent call last):
  File "build-locally.py", line 75, in <module>
    main()
  File "build-locally.py", line 71, in main
    run_docker_build(ns)
  File "build-locally.py", line 23, in run_docker_build
    subprocess.check_call([script])
  File "/home/jshimwell/miniconda3/envs/dagmc_conda_forge_build/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.
```

### CODE REVIEW

**Comment by pshriwise** [comment]
It looks like the latest version of MOAB requires a newer version of HDF5 than its `libnetcdf` dependency allows? Am I reading that right?

**Comment by shimwell** [comment]
Interesting, I can see from the moab build that they tag a specific version of libnetcdf
https://github.com/conda-forge/moab-feedstock/blob/e54bab86f6b6ddc59b4a82cf6f4792eb92192d9c/recipe/meta.yaml#L68-L69

**Comment by shimwell** [comment]
I shall try to remove ```hdf5``` and ```eigen``` from the dagmc meta.yml. These should be brought in by moab so I think we don't need them

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,12 +20,9 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
-    - hdf5
-    - moab 5.*
-    - eigen
+    - moab 5.3.1
```
Comment (shimwell) [inline recipe/meta.yaml:28]:
```suggestion
    - libnetcdf >=4.7.4
    - hdf5 >=1.12.1
    - moab 5.3.1
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -20,12 +20,9 @@ requirements:
...
   run:
-    - hdf5
-    - moab 5.*
+    - moab 5.3.1
```
Comment (shimwell) [inline recipe/meta.yaml:32]:
```suggestion
    - libnetcdf >=4.7.4
    - hdf5 >=1.12.1
    - moab 5.3.1
```

**Comment by pshriwise** [comment]
Oh no... https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=461733&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b&l=1260 😞 

Is this progress beyond the hdf5 issues though?

**Comment by shimwell** [comment]
> Oh no... https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=461733&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b&l=1260 disappointed
> 
> Is this progress beyond the hdf5 issues though?

I think this is progress but it looks like the submodules are not included in the tar.gz release file

**Comment by shimwell** [comment]
Not sure why that linux runner was canceled, it suggests a docker image was not found.

I've made use of a git repo as the source instead of the url. This can accept submodules

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-git

File: recipe/meta.yaml [hunk 3]
```diff
@@ -20,12 +21,13 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
-    - hdf5
-    - moab 5.*
-    - eigen
+    - libnetcdf >=4.7.4
+    - hdf5 >=1.12.1
```
Comment (shimwell) [inline recipe/meta.yaml:27]:
```suggestion
    - eigen
    - libnetcdf >=4.7.4
    - hdf5 >=1.12.1
```

**Comment by shimwell** [comment]
Not quite sure why the linux64 CI is failing it appears to be looking for a ubuntu-16.04 runner. However the OSx CI passed :tada:

**Comment by pshriwise** [comment]
> Not quite sure why the linux64 CI is failing it appears to be looking for a ubuntu-16.04 runner. However the OSx CI passed tada

How do we specify a newer version of Ubuntu?

**Comment by shimwell** [comment]
@conda-forge-admin, please rerender

**Comment by shimwell** [comment]
I've got this PR working and I've also got PR #6 working in the same way.

PR #6 has a cleaner commit history so closing this in favour of PR #6

**Pull request closed without merging**